docs/ql-style-guide.md

```diff
@@ -0,0 +1,434 @@
+# QL Style Guide
+
+## Introduction
+
+This document describes how to format the QL code in this repository, covering aspects such as layout, white-space, naming and documentation. Adhering to consistent standards makes code easier to read and maintain. Of course, these are only guidelines, and can be overridden as the need arises on a case-by-case basis. Where existing code deviates from these guidelines, prefer consistency with the surrounding code.
+
+Words in *italic* are defined in the [Glossary](#glossary).
+
+## Indentation
+1. *Always* use 2 spaces for indentation.
+1. *Always* indent:
+   - the *body* of a module, newtype, class or predicate,
+   - the second and subsequent lines after a line break used to split a long line,
+   - the *body* of a `from`, `where` or `select` clause where it spans multiple lines,
+   - the *body* of a *quantifier* that spans multiple lines.
+
+
+Examples:
+
+```ql
+module Helpers {
+  /** ... */
+  class X ... {
+    /** ... */
+    int getNumberOfChildren () {
+      result = count(int child |
+        exists(this.getChild(child))
+      )
+    }
+  }
+}
+```
+
+```ql
+from Call c, string reason
+where isDeprecated(c, reason)
+select c, "This call to '$@' is deprecated because " + reason + ".",
+  c.getTarget(), c.getTarget().getName()
+```
+
+## Line breaks
+1. Use UNIX line endings.
+1. Lines *must not* exceed 100 characters.
+1. Long lines *should* be split with a line break, and the following lines *must* be indented one level until the next "regular" line break.
+1. There *should* be a single blank line:
+   - between the file documentation and the first `import`,
+   - before each declaration, except for the first declaration in a *body*,
+   - before the `from`-`where`-`select` section in a query file.
+1. *Avoid* two or more adjacent blank lines.
+1. There *must* be a new line after the *annotations* `cached`, `deprecated`, `pragma`, `language` and `bindingset`.

```

Comment:
If there's more than one annotation, can they all be on the same line?

Reply:
Yes - except for these ones.